### DISCUSSION TITLE
x-cloak is blinking in v3

### DISCUSSION THREAD

**Discussion by anatoliy-t7** [alpinejs/alpine#1554] [7. Bugs (archived)] [answered]
After update to Alpine v3 x-cloak started blinkin.

See an attached examples for this issue.

I use:
Livewire: 2.5.1
Alpine: 3.0.6

**With x-cloak**
![with-x-cloak](https://user-images.githubusercontent.com/11552252/122571895-d8f55680-d06a-11eb-969f-ce528f26c799.gif)

**Without x-cloak**
![without-x-cloak](https://user-images.githubusercontent.com/11552252/122571998-f3c7cb00-d06a-11eb-81cc-7eebc5ff7a5c.gif)

**Comment by KevinBatdorf** [comment]
How and where did you define `x-cloak` in your CSS? Is this page online somewhere to see it?

The "best practice" in V2 was to place it as high up in the `<head>` of the document as you can, like this:

```html
<style>[x-cloak] { display: none!important }</style>
```

**Reply by anatoliy-t7** [reply]
Yes, it is exactly in the `<head>`.

```html
<style>[x-cloak] { display: none!important }</style>
```

**Reply by anatoliy-t7** [reply]
A similar problem #1561

**Comment by danddanddand** [comment]
I doubt this is related to x-cloak because by the time you click the edit button, page would’ve initialized and it would’ve nuked it off elements so shouldn’t come into play. 

Can you try 3.0.7 of alpine, there was change to way class bindings were happening for objects that might be in play but hard to know without seeing code.

**Reply by anatoliy-t7** [reply]
On 3.0.8 same issue.

**Comment by calebporzio** [comment, marked as answer]
Interesting, I think I have a guess why this is happening.

The new x-cloak implementation waits until the next tick of the browser to "remove" the x-cloak attribute from an element.

This is so that the entire chunk of DOM has a chance to finish initializing before x-cloak is removed and it's revealed.

I think the reason you're seeing a blip is this: Livewire is updating the template from the server and re-adding x-cloak to an element, then when it gets picked up on the front-end and initialized, it gets re-removed, but after a tick, so you see a quick blip.

After thinking about this, there is no need to wait for a full tick to remove x-cloak. I have PRd and merged a change to remove the tick wait: https://github.com/alpinejs/alpine/pull/1570

Please test this in v3.0.9 and verify it's fixed - thanks!

**Reply by anatoliy-t7** [reply]
Yes, in **v3.0.9** it's **fixed**!
Thanks!

**Reply by calebporzio** [reply]
Awesome! Can you mark this as answered please
